graph/graphs/gen/gen.go

```diff
@@ -18,3 +22,122 @@ func abs(a int) int {
 	}
 	return a
 }
+
+// NodeIDGraphBuild is a graph that can create new nodes with
+// specified IDs.
+type NodeIDGraphBuilder interface {
+	graph.Builder
+	graph.NodeWithIDer
+}
+
+// IDer is a mapping from an index to a node ID.
+type IDer interface {
+	// Len returns the length of the set of node IDs.
+	Len() int
+
+	// ID returns the ID of the indexed node.
+	// ID must be a bijective function. No check
+	// is made for this property.
+	ID(int) int64
+}
+
+// Complete constructs a complete graph in dst using nodes with the given IDs.
+func Complete(dst NodeIDGraphBuilder, ids IDer) error {
+	for i := 0; i < ids.Len(); i++ {
+		uid := ids.ID(i)
+		u, new := dst.NodeWithID(uid)
+		if new {
+			dst.AddNode(u)
```

Comment:
Are you adding the node here to cover the case of `ids.Len() == 1`? Would it make sense to separate it as in the other functions?

Reply:
Done